File: Tutorial2InjectConstructor/src/main/java/com/example/tutorial2injectconstructor/MainActivity.java
package com.example.tutorial2injectconstructor;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import com.example.tutorial2injectconstructor.model.MySharedPreferences;

import javax.inject.Inject;
import javax.inject.Named;

public class MainActivity extends AppCompatActivity {

    @Inject
    MySharedPreferences mySharedPreferences;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ((MyApplication) getApplicationContext()).getMyComponent().inject(this);

        // @ Inject calls this constructor of MySharedPreferences class after injecting SharedPreferences to it
        // SharedPreferences sharedPreferences = getApplicationContext().getSharedPreferences("PrefName", Context.MODE_PRIVATE);
        // mySharedPreferences = new MySharedPreferences(sharedPreferences);

       mySharedPreferences.putData("info", 10);

        int data = mySharedPreferences.getData("info");
        Toast.makeText(this, "DATA: " + data, Toast.LENGTH_SHORT).show();
    }


}

File: Tutorial8-3DaggerHilt-DefineComponent/src/main/java/com/smarttoolfactory/tutorial8_3daggerhilt_definecomponent/MainActivity.kt
package com.smarttoolfactory.tutorial8_3daggerhilt_definecomponent

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.smarttoolfactory.tutorial8_3daggerhilt_definecomponent.model.ApplicationObject
import dagger.hilt.EntryPoints
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject


@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

//    @Inject
//    lateinit var applicationObject: ApplicationObject

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        Toast.makeText(this, "applicationObject: $applicationObject", Toast.LENGTH_SHORT).show()

    }

}